pkg/graveler/graveler.go

```diff
@@ -1487,17 +1499,36 @@ func (g *Graveler) DeleteBatch(ctx context.Context, repository *RepositoryRecord
 	if len(keys) > DeleteKeysMaxSize {
 		return fmt.Errorf("keys length (%d) passed the maximum allowed(%d): %w", len(keys), DeleteKeysMaxSize, ErrInvalidValue)
 	}
-	var m *multierror.Error
+
 	err = g.safeBranchWrite(ctx, g.log(ctx).WithField("operation", "delete_keys"),
 		repository, branchID, func(branch *Branch) error {
-			var cachedMetaRangeID MetaRangeID // used to cache the committed branch metarange ID
+			commit, err := g.RefManager.GetCommit(ctx, repository, branch.CommitID)
+			if err != nil {
+				return err
+			}
+			cachedMetaRangeID := commit.MetaRangeID
+
+			deleteWorkers := int(math.Min(float64(100), float64(len(keys))))
+			keysChan := make(chan Key, len(keys))
+			var wg multierror.Group
+			cctx, cancel := context.WithCancel(ctx)
+			defer cancel()
+			for i := 0; i < deleteWorkers; i++ {
+				wg.Go(func() error {
+					return g.deleteConsumer(cctx, cancel, repository, branch, &cachedMetaRangeID, keysChan)
+				})
```

Comment:
I don't understand how this can return the correct errors.  We are supposed to delete _everything_ on the list, and return _all_ the errors.  But if _something_ on the list fails, AFAICT:

1. Each failing goroutine returns the _first_ error, cancels the rest, and stops.
2. Each other goroutine is _cancelled_ and returns a cancellation error for some arbitrary key -- even though that key may have been successfully deleted with no error!
3. Some keys are never tried and therefore _not_ reported as deleted and also _not_ reported as having failed deletion.

Reply:
You are right - it's part of the changes I've made while experimenting - will revert